recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -0,0 +1,889 @@
+        "with_libkml": [True, False],
+        "with_libtiff": [True, False],
+        "with_lz4": [True, False],
+        "with_mongocxx": [True, False],
+        "with_mysql": [None, "libmysqlclient", "mariadb-connector-c"],
+        "with_netcdf": [True, False],
+        "with_odbc": [True, False],
+        "with_openjpeg": [True, False],
+        "with_openssl": [True, False],
+        "with_pcre": [True, False],
+        "with_pcre2": [True, False],
+        "with_pg": [True, False],
+        "with_png": [True, False],
+        "with_podofo": [True, False],
+        "with_poppler": [True, False],
+        "with_proj": [True, False],
+        "with_qhull": [True, False],
+        "with_sqlite3": [True, False],
+        "with_webp": [True, False],
+        "with_xerces": [True, False],
+        "with_xml2": [True, False],
+        "with_zlib": [True, False],
+        "with_zstd": [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "tools": False,
+        "with_armadillo": False,
+        "with_arrow": False,
+        "with_blosc": False,
+        "with_cfitsio": False,
+        "with_crypto": False, # deprecated, replaced by with_openssl
+        "with_cryptopp": False,
+        "with_curl": False,
+        "with_dds": False,
+        "with_expat": False,
+        "with_exr": False,
+        "with_freexl": False,
+        "with_geos": True,
+        "with_gif": True,
+        "with_gta": False,
+        "with_hdf4": False,
+        "with_hdf5": False,
+        "with_heif": False,
+        "with_kea": False,
+        "with_libdeflate": True,
+        "with_libiconv": True,
+        "with_jpeg": "libjpeg",
+        "with_libkml": False,
+        "with_libtiff": True,
+        "with_lz4": False,
+        "with_mongocxx": False,
+        "with_mysql": None,
+        "with_netcdf": False,
+        "with_odbc": False,
+        "with_openjpeg": False,
+        "with_openssl": False,
+        "with_pcre": False,
+        "with_pcre2": False,
+        "with_pg": False,
+        "with_png": True,
+        "with_podofo": False,
+        "with_poppler": False,
+        "with_proj": True,
+        "with_qhull": True,
+        "with_sqlite3": True,
+        "with_webp": False,
+        "with_xerces": False,
+        "with_xml2": False,
+        "with_zlib": True,
+        "with_zstd": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```

Comment:
Needs to be a try catch here since it;s deleted above

Reply:
added try/except